### PR TITLE
Add agent_on_k8s docs and k8s provider

### PR DESCRIPTION
**Pull request by ChrsMark**
This PR adds documentation for:

1. Kubernetes Provider
2. Running Agent standalone on Kubernetes

Closes https://github.com/elastic/beats/issues/21848
Part of https://github.com/elastic/beats/issues/23613

Kubernetes provider implementation PR: https://github.com/elastic/beats/pull/21480

### CODE REVIEW

File: docs/en/ingest-management/elastic-agent/running-on-kubernetes-standalone.asciidoc [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+This manifest includes the following:
+1. Kubernetes integration to collect k8s metrics from each nodes' kubelet
+2. Kubernetes integration to collect metrics from `kube-state-metrics`. https://github.com/kubernetes/kube-state-metrics#usage[kube-state-metrics] is needed to be already deployed
+in the cluster. If `kube-state-metrics` is not already running, deploy it now (see the
+https://github.com/kubernetes/kube-state-metrics#kubernetes-deployment[Kubernetes
+deployment] docs)
+3. Kubernetes integration to collect metrics from `proxy` kubernetes component on each of the nodes.
+4. Kubernetes integration to collect metrics from Kubernetes API server, Scheduler and Controller-manager
+5. System integration to collect system level metrics from nodes.
+6. Pod's log collection using <<kubernetes-provider,dynamic inputs and kubernetes provider>>
```
[the remark this answers is not in the document]
Comment (ChrsMark) [reply docs/en/ingest-management/elastic-agent/running-on-kubernetes-standalone.asciidoc]: 👍🏼 I made it shorter and more abstract. I like the idea of linking each integration to its respective page (Kubernetes integration/package, System Integration/package etc) however I'm not sure what would be the proper page to choose. Currently we have an Integrations page (https://www.elastic.co/integrations?solution=all-solutions&category=kubernetes) but this refers to Beats integrations and not to packages 🤔 . @dedemorton any ideas here?

[the remark this answers is not in the document]
Comment (jsoriano) [reply docs/en/ingest-management/elastic-agent/running-on-kubernetes-standalone.asciidoc]: Yep, maybe this is not needed, specially thinking that this is a "running on..." page and there will be more pages with info about UIs and so on.

**Review by jsoriano — APPROVED** [review]

File: docs/en/ingest-management/elastic-agent/running-on-kubernetes-standalone.asciidoc [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+$ kubectl -n kube-system get pods -l app=elastic-agent
+NAME                            READY   STATUS    RESTARTS   AGE
+elastic-agent-4665d             1/1     Running   0          81m
+elastic-agent-9f466c4b5-l8cm8   1/1     Running   0          81m
+elastic-agent-fj2z9             1/1     Running   0          81m
+elastic-agent-hs4pb             1/1     Running   0          81m
+------------------------------------------------
+
+[float]
+==== Autodiscover targeted Pods
```
Comment (blakerouse) [inline docs/en/ingest-management/elastic-agent/running-on-kubernetes-standalone.asciidoc:104]: This is great for the docs!

**Review by blakerouse — APPROVED** [review]
Looks really good.

**Review by ph — APPROVED** [review]
Thanks @ChrsMark This is really great!

**Comment by ph** [comment]
@ChrsMark What is the release target for the doc 7.13?

**Comment by ChrsMark** [comment]
> @ChrsMark What is the release target for the doc 7.13?

@ph I think it will depend on https://github.com/elastic/observability-docs/pull/401#issuecomment-785291929 and what we will decide about the target version of manifests.

**Pull request merged**